### PR TITLE
(LSP) Report errors in files that lack a typed sigil.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
In the editor setting, report errors in files that lack a typed sigil.

The second commit adds an ENFORCE that tries to ensure that we don't repeat this mistake again.

### Motivation

Fixes https://github.com/sorbet/sorbet/issues/1187

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jvilk-stripe** [comment]
Specifically requesting review from Dmitry, since we discussed this issue previously.

File: main/pipeline/pipeline.cc [hunk 1]
```diff
@@ -167,6 +167,7 @@ ast::ParsedFile indexOne(const options::Options &opts, core::GlobalState &lgs, c
                          unique_ptr<KeyValueStore> &kvstore) {
     auto &print = opts.print;
     ast::ParsedFile dslsInlined{nullptr, file};
+    ENFORCE(file.data(lgs).strictLevel == decideStrictLevel(lgs, file, opts));
```
Comment (jvilk-stripe) [inline main/pipeline/pipeline.cc]: I don't think that it would make sense to mutate `strictLevel` in `indexOne` because that would require `indexOne` to unfreeze the file table.

If it does, though, I can move the update to `indexOne`.

File: test/test_corpus.cc [hunk 2]
```diff
@@ -165,10 +165,6 @@ TEST_P(ExpectationTest, PerPhaseTest) { // NOLINT
     vector<core::ErrorRegion> errs;
     for (auto file : files) {
         errs.emplace_back(gs, file);
-        if (file.data(ctx).source().find("# typed:") == string::npos) {
```
Comment (DarkDimius) [inline test/test_corpus.cc]: @jvilk-stripe this was intentional. Why did you remove it?

Comment (DarkDimius) [reply test/test_corpus.cc]: We want all test files to be explicit about what `typed` level they want to be in.

Comment (jvilk-stripe) [reply test/test_corpus.cc]: @DarkDimius without removing this line, I cannot test this bug...

Comment (jvilk-stripe) [reply test/test_corpus.cc]: Unless I make `# typed: none` a valid type sigil?

Comment (DarkDimius) [reply test/test_corpus.cc]: Can we hardcode name of the file? we already have prior art for this
https://github.com/sorbet/sorbet/blob/7bbc8c652d4c1ea044d01d59778beea3a5627fb1/core/Context.cc#L34

File: test/test_corpus.cc [hunk 3]
```diff
@@ -152,11 +152,16 @@ TEST_P(ExpectationTest, PerPhaseTest) { // NOLINT
...
     vector<core::FileRef> files;
+    constexpr string_view whitelistedTypedNoneTest = "missing_typed_sigil.rb"sv;
     {
         core::UnfreezeFileTable fileTableAccess(gs);
 
         for (auto &sourceFile : test.sourceFiles) {
-            files.emplace_back(gs.enterFile(test.sourceFileContents[test.folder + sourceFile]));
+            auto fref = gs.enterFile(test.sourceFileContents[test.folder + sourceFile]);
+            if (FileOps::getFileName(sourceFile) == whitelistedTypedNoneTest) {
+                fref.data(gs).strictLevel = core::StrictLevel::False;
```
Comment (jvilk-stripe) [inline test/test_corpus.cc]: `test_corpus` never runs `index` directly, so it doesn't run `decideStrictLevel`. There aren't any mock options objects, so I'm just explicitly adding this behavior for this one weird test.

This test is really intended for LSP, anyway.

**Comment by jvilk-stripe** [comment]
I believe I have addressed your comments.

**Comment by jvilk-stripe** [comment]
Also, the ENFORCE does not trigger in a test run with VS Code in a small folder I test in, so it might be just fine...
